subsys/bluetooth/host/gatt.c

```diff
@@ -933,6 +933,15 @@ void bt_gatt_foreach_attr(u16_t start_handle, u16_t end_handle,
 
 	SYS_SLIST_FOR_EACH_CONTAINER(&db, svc, node) {
 		int i;
+		struct bt_gatt_service *next;
+
+		next = SYS_SLIST_PEEK_NEXT_CONTAINER(svc, node);
+		if (next) {
+			/* Skip ahead if start is not within service handles */
+			if (next->attrs[0].handle <= start_handle) {
```

Comment:
Should this really be <= and not < ?

Reply:
It is the next service so equal still means is not in the current svc.

Reply:
right

Reply:
`next->attrs[0].handle` is the handle of the service definition of the next service? If so I don't get why you would skip ahead in that case

Reply:
Service, and its attributes, are registered in ascending order, so this is checking if the handle happens to be >= the following service which means is not withing the current service so it just skip iterating into the service handles. It is allowed to set a fixed handle, provided it is not in use, which is why we dont simple use handle + num_handles logic, though the logic is quite similar.